sycl/include/sycl/detail/pi.hpp

```diff
@@ -67,12 +67,20 @@ bool trace(TraceLevel level);
 #define __SYCL_CUDA_PLUGIN_NAME "pi_cuda.dll"
 #define __SYCL_ESIMD_EMULATOR_PLUGIN_NAME "pi_esimd_emulator.dll"
 #define __SYCL_HIP_PLUGIN_NAME "libpi_hip.dll"
-#else
+#elif defined(__SYCL_RT_OS_LINUX)
 #define __SYCL_OPENCL_PLUGIN_NAME "libpi_opencl.so"
 #define __SYCL_LEVEL_ZERO_PLUGIN_NAME "libpi_level_zero.so"
 #define __SYCL_CUDA_PLUGIN_NAME "libpi_cuda.so"
 #define __SYCL_ESIMD_EMULATOR_PLUGIN_NAME "libpi_esimd_emulator.so"
 #define __SYCL_HIP_PLUGIN_NAME "libpi_hip.so"
+#elif defined(__SYCL_RT_OS_DARWIN)
```

Comment:
Nit: I wonder if we simply need to define a suffix (like `.dll`, `.so` or `.dylib`) to use it in library names instead of duplicating almost the same lines three times

Reply:
The right way to do this is to autogenerate these names from CMake, but that'd be outside of the scope of this patch.